src/main/java/com/trainigcenter/springtask/persistence/GenreDao.java

```diff
@@ -0,0 +1,24 @@
+package com.trainigcenter.springtask.persistence;
+
+import com.trainigcenter.springtask.domain.Genre;
+import org.springframework.stereotype.Repository;
+
+import java.util.List;
+
+@Repository
+public interface GenreDao {
+
+    Genre findById(Integer id);
+
+    Genre findByName(String name);
+
+    List<Genre> findAll();
+
+    Genre add(Genre genre);
+
+    Genre update(Genre genre);
+
+    void delete(Genre genre);
```

Comment:
How does removal work? 
What if I pass `Genre` with `id` and `name` that do not match each other? 
What if I pass `Genre` with just `name` populated?